codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/JsonShapeDeserVisitor.java

```diff
@@ -190,10 +192,16 @@ protected void deserializeStructure(GenerationContext context, StructureShape sh
                     }
                     String memberName = symbolProvider.toMemberName(member);
                     String serializedMemberName = getSerializedMemberName(member);
-                    writer.openBlock("case $S:", "", serializedMemberName, () -> {
-                        String dest = "sv." + memberName;
-                        context.getModel().expectShape(member.getTarget()).accept(getMemberDeserVisitor(member, dest));
-                    });
+
+                    boolean isErrorStructure = shape.hasTrait(ErrorTrait.class);
+
+                    if(isErrorStructure && memberName.equals("Message")){
```

Comment:
```suggestion
                    if(isErrorStructure && memberName.equalsIgnoreCase("Message")){
```

Reply:
(this will handle both message and Message in the model, and of course anything in between)